src/components/Reports/FourteenMonthReports/SalaryReport/SalaryReportTable.tsx

```diff
@@ -0,0 +1,362 @@
+    backgroundColor:
+      panelopen === 1 ? theme.palette.secondary.dark : 'transparent',
+  }),
+);
+
+const NavListIcon = styled(FilterList)(({ theme }) => ({
+  width: 24,
+  height: 24,
+  color: theme.palette.primary.dark,
+}));
+
+const PrintableContainer = styled(TableContainer)(() => ({
+  // First style set size as landscape
+  // ['@media print']: {
+  //   ['@page']: { size: 'landscape' },
+  // },
+  height: 'calc(100vh - 160px)',
+}));
+
+const StickyTable = styled(Table)(({}) => ({
+  height: 'calc(100vh - 96px)',
+}));
+
+export const SalaryReportTable: React.FC<Props> = ({
+  accountListId,
+  isNavListOpen,
+  title,
+  onNavListToggle,
+  ...rest
+}) => {
+  const [isExpanded, setExpanded] = useState<boolean>(false);
+  const [order, setOrder] = useState<Order>('asc');
+  const [orderBy, setOrderBy] = useState<OrderBy | number | null>(null);
+  const reportTableRef = useRef(null);
+  const { t } = useTranslation();
+
+  const isMobile = useMediaQuery((theme: Theme) =>
+    theme.breakpoints.down('sm'),
+  );
+
+  const { data, loading, error } = useFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+      currencyType: FourteenMonthReportCurrencyType.Salary,
+    },
+  });
+
+  const handlePrint = useReactToPrint({
+    content: () => reportTableRef.current,
+  });
+
+  const contacts = useMemo(() => {
+    return data?.fourteenMonthReport.currencyGroups?.flatMap(
+      (currencyGroup) => [...currencyGroup?.contacts],
+    );
+  }, [data?.fourteenMonthReport.currencyGroups]);
+
+  console.log(contacts);
+  const orderedContacts = useMemo(() => {
+    if (contacts && orderBy) {
+      return contacts.sort((a: Contact, b: Contact) => {
+        if (orderBy === 'name' || orderBy === 'total') {
+          if (order === 'asc') {
+            return a[orderBy]
+              .toString()
+              .localeCompare(b[orderBy].toString(), undefined, {
+                numeric: true,
+              });
+          } else {
+            return b[orderBy]
+              .toString()
+              .localeCompare(a[orderBy].toString(), undefined, {
+                numeric: true,
+              });
+          }
+        } else {
+          if (typeof orderBy === 'number' && a.months && b.months) {
+            if (order === 'asc') {
+              return a['months'][orderBy]['total']
+                .toString()
+                .localeCompare(
+                  b['months'][orderBy]['total'].toString(),
+                  undefined,
+                  {
+                    numeric: true,
+                  },
+                );
+            } else {
+              return b['months'][orderBy]['total']
+                .toString()
+                .localeCompare(
+                  a['months'][orderBy]['total'].toString(),
+                  undefined,
+                  {
+                    numeric: true,
+                  },
+                );
+            }
+          }
+        }
+      });
```

Comment:
The TS error is you don't have default case. There are logic paths which don't specifically return something and therefore return `undefined`. You could add some default case at the bottom like `return 0` or something.

This is a lot of similar code. I might clean it up like this:
```suggestion
      const getSortValue = (contact: Contact) =>
        (typeof orderBy === 'number'
          ? contact['months']?.[orderBy]['total'].toString()
          : contact[orderBy]) ?? contact.name;

      return contacts.sort((a, b) => {
        const compare = getSortValue(a).localeCompare(
          getSortValue(b),
          undefined,
          {
            numeric: true,
          },
        );

        return order === 'asc' ? compare : -compare;
      });
```